pyccel/codegen/printing/pycode.py

```diff
@@ -851,29 +851,15 @@ def _print_NumpyCountNonZero(self, expr):
 
         return "{}({})".format(name, arg)
 
-    def _print_ListAppend(self, expr):
-        method_name = expr.name
+    def _print_ListMethod(self, expr):
+        method_name = self._print(expr.name)
         list_var = self._print(expr.list_variable)
-        append_arg = self._print(expr.append_argument)
-
-        return f"{list_var}.{method_name}({append_arg})\n"
-
-    def _print_ListInsert(self, expr):
-        method_name = expr.name
-        index = self._print(expr.index)
-        list_var = self._print(expr.list_variable)
-        insert_arg = self._print(expr.insert_argument)
-
-        return f"{list_var}.{method_name}({index}, {insert_arg})\n"
-
-    def _print_ListPop(self, expr):
-        args = self._print(expr.pop_index) if expr.pop_index else ""
-        name = self._print(expr.list_variable)
-        return f"{name}.pop({args})"
+        if len(expr.args) == 0 or expr.args[0] is None:
+            method_args = ''
+        else:
+            method_args = ', '.join(self._print(a) for a in expr.args)
```

Comment:
I honestly don't know if such a case exists. I still confuse None with a null pointer, thinking that it's enough to check for the first occurrence. For a more rigorous check, we can use
```python
all(arg is None for arg in expr.args):
```

Reply:
`None` is similar to a null pointer, but it would not be enough to check for the first null pointer either. Often the last value can be ignored but the first one cannot be ignored unless nothing comes after. Your rigorous check is the best I think :+1:

Reply:
fixed 0c59fc93431c998e345b81a8cabfdd68ce325fbc